libs/waterfall/src/lib/components/graph/graph.component.ts

```diff
@@ -106,7 +113,15 @@ export class WaterfallGraphComponent implements OnInit, OnDestroy {
     right.name = this.rightForm.controls.name.value;
     right.percent = this.rightForm.controls.percent.value;
     right.rightholderId = this.rightholders.find(r => r.name === this.rightForm.controls.org.value)?.id ?? '';
-    right.conditions = { operator: 'AND', conditions: this.rightForm.controls.conditions.value };
+
+    if (right.type === 'vertical') {
+      const steps = changes.updated.rights.filter(r => r.groupId === right.id);
+      this.rightForm.controls.steps.value.forEach((condtions, index) => {
+        steps[index].conditions = { operator: 'AND', conditions: condtions };
```

Comment:
maybe use `createCondition()`
```suggestion
        steps[index].conditions = { operator: 'AND', conditions: condtions.map(c => createCondition(c)) };
```

Reply:
there is no `createCondition()` function 🤔